src/core/sequence_batch_scheduler.cc

```diff
@@ -1201,6 +1202,13 @@ DirectSequenceBatch::BatcherThread(const int nice)
   while (!scheduler_thread_exit_) {
     uint64_t wait_microseconds = default_wait_microseconds;
 
+    // Wait till execution of the last enqueued payload is
+    // complete.
+    {
+      std::unique_lock<std::mutex> lk(payload_mu_);
+      payload_cv_.wait(lk, [this] { return exec_complete_; });
+    }
```

Comment:
I was just curious why waiting for completion of the previous enqueued payload leads to higher performance?

Reply:
As I understand this allows us to delay the enqueue by the maximum amount of time. This additional time allows formation of larger batches that in turn yield a greater throughput. 
@tanmayv25 feel free to correct my understanding

Reply:
Hemant is correct. If we give more time for the requests to accumulate, we can form larger batches. We wait till the Instance thread is actually done processing the payload and then only try to form the batch.
So, instead of too many small sized requests, we will have few larger batch sized requests.